src/App.tsx

```diff
@@ -15,48 +22,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const LIGHT_COLOR_STYLE = 'is-light';
+
+function preparedGoods(
+  goods: Good[],
+  { sortBy, order } : SortingParams,
+) {
+  const newGoods = [...goods];
+
+  if (sortBy) {
+    newGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.byLength:
+          return good1.length - good2.length;
+
+        case SortType.byName:
+          return good1.localeCompare(good2);
+
+        case SortType.reset:
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order === SortType.reverse) {
+    return newGoods.reverse();
+  }
+
+  return newGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortParam>('');
+  const [sortOrder, setSortOrder] = useState<ReverseParam>('');
+  const goodsForSort = preparedGoods(goodsFromServer, {
```

Comment:
You're running an expensive operation (sorting) directly in the component body. This operation will run on every render, even if the 'sortField' or 'sortOrder' haven't changed. Consider using the useMemo hook to optimize this.